web/src/inspector.js

```diff
@@ -783,6 +844,72 @@ export class Inspector extends EventTarget {
     list.append(...rows);
   }
 
+  ///  SETTINGS PAGE
+
+  _onTorrentSettingsChanged(event_) {
+    if (this.updating) {
+      return;
+    }
+
+    const { key } = event_.target.dataset;
+    const { torrents, controller } = this;
+    const val = Inspector._getValue(event_.target);
+
+    if (val === '') {
+      return;
+    }
+
+    event_.target.lastvalue = val;
+
+    const args = { [key]: val };
+    controller.editTorrents(torrents, args);
+    this.settings_dirty = true;
+  }
+
+  static _getValue(e) {
+    switch (e.type) {
+      case 'select-one':
+      case 'select-multiple':
+      case 'number':
+      case 'text':
+      case 'url': {
+        const string = e.value;
+        const n = Number(string);
+        if (!Number.isNaN(n)) {
+          return n;
+        }
+        return string;
+      }
+      default:
+        return null;
+    }
+  }
+
+  _updateSettings() {
+    const { torrents, elements } = this;
+    let limit = '';
+    let mode = '';
+    if (
+      torrents.every(
+        (v) => v.fields.seedRatioLimit === torrents[0].fields.seedRatioLimit
+      ) &&
+      torrents.length > 0
+    ) {
+      limit = torrents[0].fields.seedRatioLimit;
+    }
+    if (
+      torrents.every(
+        (v) => v.fields.seedRatioMode === torrents[0].fields.seedRatioMode
+      ) &&
+      torrents.length > 0
+    ) {
+      mode = torrents[0].fields.seedRatioMode;
+    }
```

Comment:
You could extract this to a function and call it once for seedRatioLimit and once for seedRatioMode